dianna/methods/kernelshap.py

```diff
@@ -116,6 +116,7 @@ def explain(
         self.image_segments = self._segment_image(self.input_data, n_segments,
                                                   compactness, sigma,
                                                   **slic_kwargs)
+        n_segments = np.unique(self.image_segments).size
```

Comment:
slic returns an integer mask I agree you should find the set of those to get n_segments.
https://scikit-image.org/docs/stable/api/skimage.segmentation.html#skimage.segmentation.slic

Reply:
Yes, I think I made a mistake and it is a bug here. The actual `n_segments` should be updated as the actual number of slices can be different from the given `n_segments`. Thanks for spotting it @WillemSpek ! 👍